### ISSUE TITLE
s3x - Coversion from URI to Path and Then Back to URI Loses the Initial Configuration

### ISSUE THREAD

**Issue by thsandu** [awslabs/aws-java-nio-spi-for-s3#237]
I have issues working with conversions from URI to Path and back when using the s3x and the cool endpoint overriding functionality. It seems to me that when we are using the toUri() on a path, the initial overridden configuration is lost.

I forked your repo to add a unit test.

As you can see here, the s3 case is working: https://github.com/thsandu/aws-java-nio-spi-for-s3-issues-with-s3x/blob/b22de4a8dd68745191d797225c78c035ad30abf1/src/test/java/software/amazon/nio/spi/s3/S3PathTest.java#L365

When I so something similar on the s3x case, I get a modified result: https://github.com/thsandu/aws-java-nio-spi-for-s3-issues-with-s3x/blob/b22de4a8dd68745191d797225c78c035ad30abf1/src/test/java/software/amazon/nio/spi/s3x/S3PathTest.java#L51

```
        URI uri = URI.create("s3x://somewhere.com:1010/bucket/subfolder/afile.txt");
        Path path = Paths.get(uri);

        Path convertedPath = Paths.get(path.toUri());

        then(convertedPath).isEqualTo(path);
```
The test yields:
```
expected: /subfolder/afile.txt
 but was: /afile.txt
```

**Comment by stefanofornari** [comment]
Hi @thsandu, sorry for the late reply. Thanks for reporting the issue and providing the unit test, I will look into it.

**Comment by thsandu** [comment]
> Hi @thsandu, sorry for the late reply. Thanks for reporting the issue and providing the unit test, I will look into it.

No worries, the issue with the reading of file contents from localstack was also more important for us.

**Comment by markjschreiber** [comment]
Closing with fix from @stefanofornari in #283. Feel free to re-open if this is still an issue.

**Comment by stefanofornari** [comment]
Hi @thsandu 
yes, the rational is that I wanted to keep in the URI addressing information only, while I left aside credentials as they can also take to leak of sensitive information (e.g. logging). I may be open to reconsider it if other services work differently.
